rotkehlchen/chain/ethereum/modules/eth2.py

```diff
@@ -386,24 +324,75 @@ def get_validator_daily_stats(
             to_ts=to_timestamp,
         )
 
+    def add_validator(self, validator_index: Optional[int], public_key: Optional[Eth2PubKey]) -> None:  # noqa: E501
+        """Adds the given validator to the DB. Due to marshmallow here at least one
+        of the two arguments is not None.
+
+        May raise:
+        - RemoteError if there is a problem with querying beaconcha.in for more info
+        - InputError if the validator is already in the DB
+        """
+        valid_index: int
+        valid_pubkey: Eth2PubKey
+        dbeth2 = DBEth2(self.database)
+        if validator_index is not None and public_key is not None:
+            field = 'validator_index'
+            valid_index = validator_index
+            valid_pubkey = public_key
+            if dbeth2.validator_exists(field=field, arg=valid_index):
+                raise InputError(f'Validator {valid_index} already exists in the DB')
+        else:  # we are missing one of the 2
+            if validator_index is None:
+                field = 'public_key'
+                arg = public_key
+            else:  # we should have valid index
+                field = 'validator_index'
+                arg = validator_index  # type: ignore
+
+            if dbeth2.validator_exists(field=field, arg=arg):  # type: ignore
+                raise InputError(f'Validator {arg} already exists in the DB')
+
+            # at this point we gotta query for one of the two
+            result = self.beaconchain._query(
+                module='validator',
+                endpoint=None,
+                encoded_args=arg,  # type: ignore
+            )
+            if not isinstance(result, dict):
+                raise RemoteError(
+                    f'Validator data for {arg} could not be found. Likely invalid validator.')
+
+            try:
+                valid_index = result['validatorindex']
+                valid_pubkey = Eth2PubKey(result['pubkey'])
+            except KeyError as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key entry for {msg}.'
+
+                raise RemoteError(f'Failed to query beaconcha.in for validator data due to: {msg}') from e  # noqa: E501
+
+        # by now we have a valid index and pubkey. Add to DB
+        dbeth2.add_validators([Eth2Validator(index=valid_index, public_key=valid_pubkey)])
+
     # -- Methods following the EthereumModule interface -- #
     def on_startup(self) -> None:
         pass
 
-    def on_account_addition(self, address: ChecksumEthAddress) -> Optional[List['AssetBalance']]:
+    def on_account_addition(self, address: ChecksumEthAddress) -> Optional[List[AssetBalance]]:  # pylint: disable=useless-return  # noqa: E501
+        """Just query balances and add detected validators to DB. Return nothing"""
         try:
-            result = self.get_balances([address])
+            self.get_balances(
```

Comment:
Unless I'm missing something in the diff get_balances returns the balances per address and doesn't modify any internal variable to the class. Should you return the result of this call?

Reply:
the ethereum modules expects map of addresses. This is a map of validators. Can't work with it.